### PR TITLE
Add rejected flag to case database

### PR DESCRIPTION
**Pull request by lucas-shaw**
## Description
Add a flag to the case model which confirms whether a case was ever rejected. To be used for reporting purposes. 

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
https://dsdmoj.atlassian.net/jira/software/c/projects/CDPT/boards/1152?selectedIssue=CDPT-1207

### CODE REVIEW

File: app/models/case/sar/offender.rb [hunk 1]
```diff
@@ -95,6 +95,7 @@ def searchable_fields_and_ranks
                  is_partial_case: :boolean,
                  partial_case_letter_sent_dated: :date,
                  further_actions_required: :string,
+                 case_originally_rejected: :boolean,
```
Comment (lucas-shaw) [inline app/models/case/sar/offender.rb:98]: I didn't add a ? to the attr name as the other booleans don't seem to.

File: app/models/case/sar/offender.rb [hunk 2]
```diff
@@ -459,6 +461,10 @@ def use_subject_as_requester
     self.name = subject_full_name
   end
 
+  def set_case_originally_rejected
+    self.case_originally_rejected = true
```
Comment (lucas-shaw) [inline app/models/case/sar/offender.rb:465]: I assumed it would be ok to not give a value as false if its not a rejected case. So either nil or true.

File: spec/models/case/sar/offender_spec.rb [hunk 3]
```diff
@@ -290,6 +290,13 @@
     end
   end
 
+  describe "when creating a rejected case" do
```
Comment (lucas-shaw) [inline spec/models/case/sar/offender_spec.rb:293]: Thought about adding a test to check if it returns nil for a non rejected case, but not sure whether thats necessary.

**Review by vertism — APPROVED** [review]
LGTM
